pulsar-connect/core/src/main/java/org/apache/pulsar/connect/core/Source.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.connect.core;
+
+import java.util.Map;
+
+public interface Source<T> extends AutoCloseable {
+    /**
+     * Open connector with configuration
+     *
+     * @param config initialization config
+     * @throws Exception IO type exceptions when opening a connector
+     */
+    void open(final Map<String, String> config) throws Exception;
+
+    /**
+     * Reads the next message from source
+     * @return next message from source
+     * @throws Exception
```

Comment:
can you clarify if this is a blocking operation or if there is nothing found, T can be null?

Reply:
I will add comments